app/services/SubscribeService.scala

```diff
@@ -106,16 +106,42 @@ trait SubscribeService {
   private def createEnrolmentVerifiers(response: HttpResponse, data: JsValue): Verifiers = {
     getUtrAndPostCode(data) match {
       case (Some(uniqueTaxRef), Some(ukClientPostCode)) =>
-        Verifiers(List(
-          Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
-          Verifier(GovernmentGatewayConstants.VerifierCtUtr, uniqueTaxRef))
-        )
+        (data \ "businessType").asOpt[String] match {
+          case Some("LLP") => {
+            Verifiers(List(
+              Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
+              Verifier(GovernmentGatewayConstants.VerifierSaUtr, uniqueTaxRef))
+            )
+          }
+          case Some("Corporate Body") => {
+            Verifiers(List(
+              Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
+              Verifier(GovernmentGatewayConstants.VerifierCtUtr, uniqueTaxRef))
+            )
+          }
+          case Some("Partnership") => {
+            Verifiers(List(
+              Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
+              Verifier(GovernmentGatewayConstants.VerifierSaUtr, uniqueTaxRef))
+            )
+          }
```

Comment:
Do you need a _ => case as well in case it's none of the above?

To also solve the cyclometic complexity I would recommend something like this instead...

  private def createEnrolmentVerifiers(response: HttpResponse, data: JsValue): Verifiers = {

    val utrType = (data \ "businessType").asOpt[String] match {
      case Some("LLP") | Some("Partnership") => GovernmentGatewayConstants.VerifierSaUtr
      case _ => GovernmentGatewayConstants.VerifierCtUtr
    }

    getUtrAndPostCode(data) match {
      case (Some(uniqueTaxRef), Some(ukClientPostCode)) =>
            Verifiers(List(
              Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
              Verifier(utrType, uniqueTaxRef))
            )
      case (None, Some(nonUkClientPostCode)) =>
        Verifiers(List(
          Verifier(GovernmentGatewayConstants.VerifierNonUKPostalCode, nonUkClientPostCode))
        ) //N.B. Non-UK Clients might use the property UK Postcode or their own Non-UK Postal Code
      case (Some(uniqueTaxRef), None) =>
            Verifiers(List(Verifier(utrType, uniqueTaxRef)))
      case (_, _) =>
        throw new RuntimeException(s"[NewRegisterUserService][subscribeAted][createEMACEnrolRequest] - postalCode or utr must be supplied")
    }
  }

Reply:
Thanks for this, I've replaced the previous code block with this one.